crates/flox-rust-sdk/src/models/environment/mod.rs

```diff
@@ -377,9 +382,19 @@ pub fn lock_manifest(
         pkgdb_cmd.arg(canonical_lockfile_path);
     }
     let output = pkgdb_cmd.output().map_err(EnvironmentError2::PkgDbCall)?;
-    let lockfile_json: Value =
-        serde_json::from_slice(&output.stdout).map_err(EnvironmentError2::ParseLockfileJSON)?;
-    Ok(lockfile_json)
+    if !output.status.success() {
+        if let Ok::<PkgDbError, _>(pkgdb_err) = serde_json::from_slice(&output.stdout) {
+            Err(EnvironmentError2::LockManifest(pkgdb_err))
+        } else {
+            Err(EnvironmentError2::ParsePkgDbError(
+                String::from_utf8_lossy(&output.stdout).to_string(),
+            ))
+        }
+    } else {
```

Comment:
nit: some comment explaining this would be nice

Reply:
Added a comment, let me know if that clarifies what was unclear